nucleus/common/common-util/src/main/java/com/sun/enterprise/util/io/FileUtils.java

```diff
@@ -551,8 +551,52 @@ else if (!deleteFile(f) && undeletedFiles != null) {
      * @param f file to delete
      */
     public static void deleteFileNowOrLater(File f) {
-        if(!deleteFile(f))
-            f.deleteOnExit();
+        if(!deleteFile(f)) {
+            deleteOnExitRecursively(f);
+        }
+    }
+
+    /**
+     * Deletes the provided file by registering a shutdown hook with the
+     * Java Runtime. Non-empty directories will be deleted recursively.
+     * @param f the file to delete on JVM shutdown
+     */
+    public static void deleteOnExitRecursively(File f) {
```

Comment:
Is it ok to delete recursively files that are not directories?

Reply:
Sure. The same way it is ok to iterate a list with one item. But I see the wording problem.